### PR TITLE
Integration test and n+1 problem

### PR DESCRIPTION
**Pull request by theBwoyProgrammer**
### BLOG APP: Integration specs for views and fix n+1 problems

Ensured that the N+1 problem is solved when fetching all posts and their comments for a user. 
Used Capybara to write integration tests for each view in the project.

**In User index page:**

-  Could see the username of all other users.
-  Could see the profile picture for each user.
-  Could see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.
- 

**In user show page:**

-  Could see the user's profile picture.
-  Could see the user's username.
-  Could see the number of posts the user has written.
-  Could see the user's bio.
-  Could see the user's first 3 posts.
-  Could see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.
- 

**User post index page:**

- could see the user's profile picture.
- could see the user's username.
- Could see the number of posts the user has written.
- Could see a post's title.
- Could see some of the post's body.
- Could see the first comments on a post.
- Could see how many comments a post has.
- Could see how many likes a post has.
- Could see a section for pagination if there are more posts than fit on the view.
- When clicked on a post, it redirects to that post's show page.

**In Post show page:**

-  Could see the post's title.
- Could see who wrote the post.
-  Could see how many comments it has.
-  Could see how many likes it has.
-  Could see the post body.
-  Could see the username of each commentor.
-  Could see the comment each commentor left.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ Metrics/ClassLength:
   Max: 150
 Metrics/BlockLength:
   IgnoredMethods: ['describe']
-  Max: 30
+  Max: 200
```
Comment (descholar-ceo) [inline .rubocop.yml:29]: - [x] Don't make changes in the rubocop config file

File: .rubocop.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ AllCops:
   DisplayCopNames: true
 
 Layout/LineLength:
-  Max: 120
+  Max: 300
```
Comment (descholar-ceo) [inline .rubocop.yml:14]: - [ ] Don't make changes in the rubocop config file

**Review by descholar-ceo — APPROVED** [review]
Hi @techwolf404, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all of the requirements ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: .rubocop.yml (hunk 1 above)
Comment (theBwoyProgrammer) [reply .rubocop.yml:29]: ok, thanks

**Pull request merged**